### PR TITLE
chore: add release workflow

### LINKED ISSUE #54: add workflow for publishing
This would streamline publishing new versions of the package to npm without depending on permissions, we could base the workflow on [this](https://github.com/digidem/mapeo-schema/blob/main/.github/workflows/release.yml)

### PR DESCRIPTION
**Pull request by tomasciccola**
this should close #54

### CODE REVIEW

**Comment by tomasciccola** [comment]
missing for this is adding the npm token to the project

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - uses: nearform-actions/optic-release-automation-action@v4
+        with:
+          commit-message: 'Release {version}'
+          sync-semver-tags: true
+          access: 'public'
+          # This prefix is added before the prerelease number, e.g. `v3.0.0-alpha.0`
+          prerelease-prefix: 'alpha'
+          semver: ${{ github.event.inputs.semver }}
+          # Prereleases are published under the `next` npm dist-tag
+          npm-tag: ${{ startsWith(github.event.inputs.semver, 'pre') && 'alpha' || 'latest' }}
```
Comment (EvanHahn) [inline .github/workflows/release.yml:44]: This comment doesn't match the behavior. I think we should use the `next` dist-tag.

```suggestion
          npm-tag: ${{ startsWith(github.event.inputs.semver, 'pre') && 'next' || 'latest' }}
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          # Don't notify linked issues
+          notify-linked-issues: false
+          # optional: set this secret in your repo config for publishing to NPM
+          npm-token: ${{ secrets.NPM_TOKEN }}
+          build-command: |
+            npm install
```
Comment (EvanHahn) [inline .github/workflows/release.yml:50]: This should use `npm ci`. I opened [a PR to do the same on `mapeo-core-next` too](https://github.com/digidem/mapeo-core-next/pull/694).

```suggestion
            npm ci
```

**Review by EvanHahn — CHANGES_REQUESTED** [review]
LGTM other than my two nitpicks.

**Review by EvanHahn — APPROVED** [review]

**Pull request merged**